sqlmesh/core/engine_adapter/spark.py

```diff
@@ -321,12 +321,28 @@ def _create_table_properties(
                 )
             )
 
-        for key, value in (table_properties or {}).items():
-            properties.append(exp.Property(this=key, value=value))
+        properties.extend(self.__table_properties_to_expressions(table_properties))
 
         if properties:
             return exp.Properties(expressions=properties)
         return None
 
+    def _create_view_properties(
+        self,
+        table_properties: t.Optional[t.Dict[str, exp.Expression]] = None,
+    ) -> t.Optional[exp.Properties]:
+        """Creates a SQLGlot table properties expression for view"""
+        if not table_properties:
+            return None
+        return exp.Properties(expressions=self.__table_properties_to_expressions(table_properties))
+
+    @classmethod
+    def __table_properties_to_expressions(
+        cls, table_properties: t.Optional[t.Dict[str, exp.Expression]] = None
+    ) -> t.List[exp.Property]:
+        if not table_properties:
+            return []
+        return [exp.Property(this=key, value=value) for key, value in table_properties.items()]
```

Comment:
I think this will mutate `value` by changing its `parent` field, although not sure if it's a problem in this case.

Reply:
Is there a more recommended way to do this?

Reply:
If we need to avoid mutations here, then you can instead do `value=value.copy()`.

To clarify, the mutation happens inside `Expression`'s constructor, due to the `_set_parent` method:

https://github.com/tobymao/sqlglot/blob/baab165f04714d443bb5faced5a1a139f8eaf81c/sqlglot/expressions.py#L98-L100